src/JWT/Builder/JwtBuilder.cs

```diff
@@ -247,6 +249,22 @@ public string Encode()
             return _encoder.Encode(_jwt.Header, _jwt.Payload, _secrets?[0]);
         }
 
+        public string Encode<T>(T payload)

```

Comment:
Should this be just a wrapper and the actual implementation be in JwtEncoder or rather IJwtEncoderExceptions? So it won't be mandatory to the builder, so far it was not.

Reply:
I'm not sure, I guess it's good to have the option to use either version

Reply:
I don't follow exactly how you want it

Reply:
Made the change, check it out. Does it make sense to you?

Reply:
Yeah, i understand now 😀

We should have a test for it to see what happens if you send Encode(type, payload) where they are different types.

Reply:
added test